### PR TITLE
Integrating with travis-ci

### PR DESCRIPTION
**Pull request by neoword**
* Fixes Issue #6
* Integrates with travis-ci

## Added
* Integration with travis-ci 

# PR Checklist Forms

- [x] CHANGELOG.md updated
- [x] Reviewer assigned
- [x] PR assigned (presumably to submitter)
- [x] Labels added (enhancement, bug, documentation) 
- [x] Milestone selected
- [x] Any annotations for reviewers have been added
      (with preference of annotating directly in code comments)
- [x] PR builds successfully

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+script: make build
+
+deploy:
+  skip_cleanup: true
+  script: make deploy
+
+notifications:
+  email:
+    recipients:
```
Comment (prabhakar1983) [inline .travis.yml]: How to become the member of this group notification `stream-registry-committers@groups.homeawaycorp.com ?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+language: java
+
+dist: trusty
+
+jdk:
+  - oraclejdk8
+
```
Comment (neoword) [inline .travis.yml]: add comment here why this is necessary

File: pom.xml [hunk 3]
```diff
@@ -539,6 +510,9 @@
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-deploy-plugin</artifactId>
                     <version>${deploy.pluginVersion}</version>
```
Comment (neoword) [inline pom.xml]: comment why this is disabled

File: pom.xml [hunk 4]
```diff
@@ -671,64 +632,56 @@
...
+            <id>ossrh</id>
+            <url>https://oss.sonatype.org/content/repositories/snapshots</url>
         </snapshotRepository>
-
-        <!-- This is needed so that site-deploy doesn't freak out. We will be using
-            gh-pages instead. -->
-        <site>
-            <id>site-deploy-noop</id>
-            <url>file:${project.build.directory}/${project.distributionManagement.site.id}</url>
-        </site>
```
Comment (neoword) [inline pom.xml]: Please put this back in here.

File: pom.xml [hunk 5]
```diff
@@ -671,64 +633,66 @@
...
         </repository>
+    </repositories>
 
+    <profiles>
+        <profile>
+            <!-- We create a profile here for two things: -->
+            <!-- 1. centrally specify the deploy to ossrh -->
+            <!-- 2. Sign all relevant artifacts (via configured GPG keys) -->
+            <!-- TODO: Document this process -->
+            <id>oss</id>
```
[the remark this answers is not in the document]
Comment (neoword) [reply pom.xml]: on it

**Pull request merged**